jira init fails when selecting default board for on-premise install

**Describe the bug**
I ran `jira init` to connect to my JIRA on-premise server, however after selecting the default board, the program panicked.

**Please provide following details**
1. JiraCLI Version:
    ```
    (Version="0.1.0", GitCommit="b2f09ec08a3d277de9e3530621572184de2ecf7a", GoVersion="go1.17.2", BuildDate="2021-10-31"", Compiler="gc", Platform="darwin/amd64")
    ```
2. Are you using Jira cloud or on-premise jira server?
    ```
    On-premise
    ```
3. What operating system are you using? Also mention version.
    ```
    macOS Monterey 12.0.1 (21A559)
    ```
4. What terminal are you using? Also mention version.
    ```
    alacritty 0.9.0 (fed349a)
    ```

**To Reproduce**

Steps to reproduce the behavior:
1. Export JIRA password to `JIRA_API_TOKEN` shell variable
2. Run jira init and follow prompts
3. See error after selecting default board

**Expected behavior**
I have not used jira-cli before, but I would expect it would finish setup and allow me to run further commands

**Screenshots**
![image](https://user-images.githubusercontent.com/11057501/139708459-4bd1823f-4ae6-4f6e-9864-20cff525b15d.png)


**Additional context**
This might depend on the board setup, but I'm not sure how to provide that to you. We are using a Kanban board setup
Panic information here:
```
panic: interface conversion: interface {} is nil, not string

goroutine 1 [running]:
github.com/ankitpokhrel/jira-cli/internal/config.(*JiraCLIConfig).configureMetadata(0xc0006fdcf8)
	/Users/ankitpokhrel/go/src/github.com/ankitpokhrel/jira-cli/internal/config/generator.go:303 +0x651
github.com/ankitpokhrel/jira-cli/internal/config.(*JiraCLIConfig).Generate(0x17c2860)
	/Users/ankitpokhrel/go/src/github.com/ankitpokhrel/jira-cli/internal/config/generator.go:79 +0x74
github.com/ankitpokhrel/jira-cli/internal/cmd/init.initialize(0xc0006b1900, {0x1855e8e, 0x0, 0x0})
	/Users/ankitpokhrel/go/src/github.com/ankitpokhrel/jira-cli/internal/cmd/init/init.go:29 +0x65
github.com/spf13/cobra.(*Command).execute(0xc0006b1900, {0x1f9deb0, 0x0, 0x0})
	/Users/ankitpokhrel/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:856 +0x5f8
github.com/spf13/cobra.(*Command).ExecuteC(0xc0006b1680)
	/Users/ankitpokhrel/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960 +0x3ad
main.main()
	/Users/ankitpokhrel/go/src/github.com/ankitpokhrel/jira-cli/cmd/jira/main.go:12 +0x1e
```

--- a/internal/config/generator.go
+++ b/internal/config/generator.go
@@ -297,11 +297,6 @@ func (c *JiraCLIConfig) configureMetadata() error {
 }
 
 func (c *JiraCLIConfig) decipherEpicField(epicMeta map[string]interface{}) string {
-	const (
-		nameField = "Epic Name"
-		linkField = "Epic Link"
-	)
-
 	var epicKey string
 
 	for field, value := range epicMeta {
@@ -309,22 +304,19 @@ func (c *JiraCLIConfig) decipherEpicField(epicMeta map[string]interface{}) strin
 			continue
 		}
 		v := value.(map[string]interface{})
-		name := v["name"].(string)
+
+		if v["name"].(string) != "Epic Name" {
+			continue
+		}
 
 		switch c.value.installation {
 		case jira.InstallationTypeCloud:
-			if name == nameField {
-				epicKey = v["key"].(string)
-			}
+			epicKey = v["key"].(string)
 		case jira.InstallationTypeLocal:
-			if name == linkField {
-				if _, ok := v["fieldId"]; !ok {
-					epicKey = field
-				}
-			} else if name == nameField {
-				if _, ok := v["fieldId"]; ok {
-					epicKey = v["fieldId"].(string)
-				}
+			if _, ok := v["fieldId"]; ok {
+				epicKey = v["fieldId"].(string)
+			} else {
+				epicKey = field
 			}
 		}
 	}

--- a/internal/config/generator.go
+++ b/internal/config/generator.go
@@ -275,10 +275,10 @@ func (c *JiraCLIConfig) configureMetadata() error {
 	}
 
 	var (
-		epicMeta map[string]interface{}
-		epicKey  string
+		epicMeta   map[string]interface{}
+		issueTypes = make([]*jira.IssueType, 0, len(meta.Projects[0].IssueTypes))
 	)
-	issueTypes := make([]*jira.IssueType, 0, len(meta.Projects[0].IssueTypes))
+
 	for _, it := range meta.Projects[0].IssueTypes {
 		if it.Name == jira.IssueTypeEpic {
 			epicMeta = it.Fields
@@ -289,25 +289,47 @@ func (c *JiraCLIConfig) configureMetadata() error {
 			Subtask: it.Subtask,
 		})
 	}
+
 	c.value.issueTypes = issueTypes
+	c.value.epic = &jira.Epic{Field: c.decipherEpicField(epicMeta)}
+
+	return nil
+}
+
+func (c *JiraCLIConfig) decipherEpicField(epicMeta map[string]interface{}) string {
+	const (
+		nameField = "Epic Name"
+		linkField = "Epic Link"
+	)
+
+	var epicKey string
 
 	for field, value := range epicMeta {
 		if !strings.Contains(field, "customfield") {
 			continue
 		}
 		v := value.(map[string]interface{})
-		if v["name"].(string) == "Epic Name" {
-			if c.value.installation == jira.InstallationTypeCloud {
+		name := v["name"].(string)
+
+		switch c.value.installation {
+		case jira.InstallationTypeCloud:
+			if name == nameField {
 				epicKey = v["key"].(string)
-			} else if c.value.installation == jira.InstallationTypeLocal {
-				epicKey = v["fieldId"].(string)
 			}
-			break
+		case jira.InstallationTypeLocal:
+			if name == linkField {
+				if _, ok := v["fieldId"]; !ok {
+					epicKey = field
+				}
+			} else if name == nameField {
+				if _, ok := v["fieldId"]; ok {
+					epicKey = v["fieldId"].(string)
+				}
+			}
 		}
 	}
-	c.value.epic = &jira.Epic{Field: epicKey}
 
-	return nil
+	return epicKey
 }
 
 func (c *JiraCLIConfig) write() error {

